xdp/common.h

```diff
@@ -1,9 +1,125 @@
 #pragma once
 
-#define PACKET_SIZE 2048
+#include <linux/types.h>
+#define __unused __attribute__ ((unused))
+
+#define LIKELY(x) (__builtin_expect (!!(x), 1))
+#define UNLIKELY(x) (__builtin_expect (!!(x), 0))
+
+#define BARRIER() __asm__ __volatile__ ("" ::: "memory")
+
+#define BUSY_WAIT(cond) \
+    do                  \
+    {                   \
+        BARRIER ();     \
+    } while (cond)
+
+#ifndef DEBUG
+#define DEBUG 0
+#endif
+
+#if DEBUG
+#define LOG(stream, fmt, ...)                 \
+    do                                        \
+    {                                         \
+        fprintf (stream, fmt, ##__VA_ARGS__); \
+        fflush (stream);                      \
+    } while (0)
+
+#define PERROR(errno)   \
+    do                  \
+    {                   \
+        perror (errno); \
+    } while (0)
+#else
+#define LOG(stream, fmt, ...)
+#define PERROR(errno)
+#endif
+
+#define START_TIMER() uint64_t __start = get_time_ns ()
+#define STOP_TIMER()                             \
+    do                                           \
+    {                                            \
+        uint64_t __end = get_time_ns ();         \
+        printf ("Time: %lu\n", __end - __start); \
+        fflush (stdout);                         \
+    } while (0)
+
+#define TIME_CODE(code)                      \
+    do                                       \
+    {                                        \
+        uint64_t start = get_time_ns ();     \
+        (code);                              \
+        uint64_t end = get_time_ns ();       \
+        printf ("Time: %lu\n", end - start); \
+        fflush (stdout);                     \
+    } while (0)
+
+/**
+ * Size of the whole pingpong packet.
+ * Although the actual payload (Ethernet header + IP header + pingpong payload) is smaller than this,
+ * the packet should be the same across all the XDP, RDMA and DPDK measurements.
+ */
+#define PACKET_SIZE 1024
+
+/**
+ * Size of the exchanged packet containing MAC address (6 bytes) and IP address (4 bytes) of each machine.
+ * This packet is sent before the start of pingpong to exchange the addresses without hardcoding them
+ */
+#define INFO_PACKET_SIZE (ETH_ALEN + sizeof (uint32_t))
+
+// Custom ethernet protocol number
+#define ETH_P_PINGPONG 0x2002
```

Comment:
ethertype is sometimes used for VLANs, we had some issues with it in some experiments where the switch was dropping packets (I don't recall why or for what network configuration). Let's keep it in mind in case something similar happens